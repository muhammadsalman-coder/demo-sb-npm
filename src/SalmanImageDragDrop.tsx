/* eslint-disable max-len */

import { Box, styled, Typography } from '@mui/material';
import React from 'react';
import CameraAltIcon from '@mui/icons-material/CameraAlt';

interface filewrapperInterface {
  error?: boolean;
  height?: string;
  minHeight?: string;
}
const FileWrapper = styled(Box)(
  ({ error, height, minHeight }: filewrapperInterface) => ({
    position: 'relative',
    display: 'flex',
    color: 'grey',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    padding: '1px',
    height: `${height ? height : '204px'}`,
    minHeight: minHeight,
    marginTop: '16px',
    borderRadius: '5px',
    overflow: 'hidden',
    border: `2px dashed ${error ? 'red' : 'grey'}`,

    '&:hover': {
      borderColor: `${error ? 'red' : 'blue'}`,
    },
    '& input': {
      position: 'absolute',
      top: 0,
      left: 0,
      fontSize: '400px',
      cursor: 'pointer',
      opacity: 0,
    },
  })
);
const Image = styled('img')({
  width: '100%',
  height: '100%',
  objectFit: 'contain',
});
export interface SalmanDragDropInterface {
  onchange: (file: any) => void;
  helperText?: string;
  error?: boolean;
  height?: string;
  minHeight?: string;
}

const IMAGE_TYPES = ['image/jpg', 'image/jpeg', 'image/png'];

const SalmanImageDragDrop = ({
  onchange,
  onBlur,
  helperText,
  error = false,
  minHeight,
  height = '204px',
}: SalmanDragDropInterface | any) => {
  const [mediaFile, setMediaFile] = React.useState<any>();

  const handleFile = (e: any) => {
    if (onchange) {
      onchange(e.target.files[0]);
    }
    if (IMAGE_TYPES.includes(e.target.files[0].type)) {
      if (e.target.files.length !== 0) {
        const reader = new FileReader();
        reader.onload = (ee: any) => {
          setMediaFile(ee.target.result);
        };
        reader.readAsDataURL(e.target.files[0]);
      }
    }
  };
  return (
    <div
      className="salmanImageDragDrop-root"
      style={{ fontFamily: "'roboto',sans-serif" }}
    >
      <label htmlFor="SalmandragdropFileinput">Image Here:</label>
      <FileWrapper error={error} height={height} minHeight={minHeight}>
        <input
          type="file"
          className="SalmandragdropFile-input"
          id="SalmandragdropFileinput"
          accept="image/png, image/jpg, image/jpeg"
          onChange={handleFile}
          onBlur={onBlur ? onBlur : null}
        />

        {mediaFile ? (
          <Image src={mediaFile} alt={'img'} />
        ) : (
          <>
            <Box>
              <CameraAltIcon />
              {/* <img src={iconImage} width={40} alt="icon" /> */}
            </Box>
            <Box>
              <Typography variant="body1" color="initial">
                Drag and Drop Image Here or
                <Typography variant="body1" color="primary" component={'span'}>
                  Click to Browse
                </Typography>
              </Typography>
              <ul>
                <li>
                  <p>JPG, PNG</p>
                </li>
                <li>
                  <p>Image will be cropped in 3:1 ratio</p>
                </li>
              </ul>
            </Box>
          </>
        )}
      </FileWrapper>
      {error && helperText && (
        <span style={{ paddingLeft: '10px', color: 'red' }}>{helperText}</span>
      )}
    </div>
  );
};

export { SalmanImageDragDrop };
