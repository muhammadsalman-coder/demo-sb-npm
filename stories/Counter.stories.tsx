import React from 'react';
import { Meta, Story } from '@storybook/react';
import { SalmanCounter, SalmanPropsInterface } from '../src/SalmanCounter';
const meta: Meta = {
  title: 'Counter',
  component: SalmanCounter,
  argTypes: {
    onchange: { action: 'onchange' },
  },
};
// argTypes: {
//   children: {
//     control: {
//       type: 'text',
//     },
//   },
// },
// parameters: {
//   controls: { expanded: true },
// },

export default meta;

const Template: Story<SalmanPropsInterface> = args => (
  <SalmanCounter {...args} />
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({
  children: <></>,
});

Default.args = {
  yourName: 'Shan',
};
