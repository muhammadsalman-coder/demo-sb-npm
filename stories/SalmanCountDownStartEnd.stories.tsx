import React from 'react';
import { Meta, Story } from '@storybook/react';
import { SalmanCountDownStartEnd } from '../src/SalmanCountDownStartEnd';

const meta: Meta = {
  title: 'SalmanCountDownStartEnd',
  component: SalmanCountDownStartEnd,
  argTypes: {
    endDateTime: { control: 'date' },
    startDateTime: { control: 'date' },
    color: { control: 'color' },
  },
};
// argTypes: {
//   children: {
//     control: {
//       type: 'text',
//     },
//   },
// },
// parameters: {
//   controls: { expanded: true },
// },

export default meta;

const Template: Story<any> = args => <SalmanCountDownStartEnd {...args} />;

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});

Default.args = {
  timerEndText: 'ended Timer',
  timerStartText: 'Start on: ',
  gap: 30,
  color: '#000',
  timeHeadinSize: '26px',
  timeTextSize: '16px',
  timerEndSize: '16px',
};
