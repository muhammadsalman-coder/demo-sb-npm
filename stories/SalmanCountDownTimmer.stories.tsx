import React from 'react';
import { Meta, Story } from '@storybook/react';
import { SalmanCountDownTimmer } from '../src/SalmanCountDownTimmer';

const meta: Meta = {
  title: 'SalmanCountDownTimmer',
  component: SalmanCountDownTimmer,
  argTypes: {
    endDateTime: { control: 'date' },
    startDateTime: { control: 'date' },
    color: { control: 'color' },
  },
};
// argTypes: {
//   children: {
//     control: {
//       type: 'text',
//     },
//   },
// },
// parameters: {
//   controls: { expanded: true },
// },

export default meta;

const Template: Story<any> = args => <SalmanCountDownTimmer {...args} />;

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});

Default.args = {
  timerEndText: 'ended Timer',
  gap: 30,
  color: '#000',
  timeHeadinSize: '26px',
  timeTextSize: '16px',
  timerStartText: 'Timer Start on: ',
  timerStartSize: '16px',
  timerEndSize: '16px',
};
