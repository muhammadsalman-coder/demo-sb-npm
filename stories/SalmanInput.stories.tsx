import React from 'react';
import { Meta, Story } from '@storybook/react';
import { SalmanInput, SalmanInputInterface } from '../src/SalmanInput';
const meta: Meta = {
  title: 'Input',
  component: SalmanInput,
};
// argTypes: {
//   children: {
//     control: {
//       type: 'text',
//     },
//   },
// },
// parameters: {
//   controls: { expanded: true },
// },

export default meta;

const Template: Story<SalmanInputInterface | any> = args => (
  <SalmanInput {...args} onChange={console.log} />
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  borderRadius: '20px',
  type: 'text',
  placeholder: 'enter name',
  padding: '10px',
  width: '100%',
};
export const TextInput = Template.bind({});
TextInput.args = {
  borderRadius: '20px',
  type: 'text',
  placeholder: 'enter name',
  padding: '10px',
  width: '100%',
};
export const PasswordInput = Template.bind({});
PasswordInput.args = {
  borderRadius: '20px',
  type: 'password',
  placeholder: 'enter password',
  padding: '10px',
  width: '100%',
};
export const FileInput = Template.bind({});
FileInput.args = {
  borderRadius: '20px',
  type: 'file',
  placeholder: 'enter password',
  padding: '10px 10px 10px 35px',
  width: 'fit-content',
};
