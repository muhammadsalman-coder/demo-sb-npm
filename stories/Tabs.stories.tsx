import React from 'react';
import { Meta, Story } from '@storybook/react';
import { SalmanTabs, ItemInterface } from '../src/SalmanTabs';
import HomeIcon from '@mui/icons-material/Home';
const meta: Meta = {
  title: 'Tabs',
  component: SalmanTabs,
};
// argTypes: {
//   children: {
//     control: {
//       type: 'text',
//     },
//   },
// },
// parameters: {
//   controls: { expanded: true },
// },

export default meta;

const Template: Story<ItemInterface> = args => (
  <SalmanTabs
    item={[
      { text: 'Home', icon: <HomeIcon /> },
      { text: 'About', icon: <HomeIcon /> },
      { text: 'Contact', icon: <HomeIcon /> },
      { text: 'Home', icon: <HomeIcon /> },
      { text: 'About', icon: <HomeIcon /> },
      { text: 'Contact', icon: <HomeIcon /> },
      { text: 'Home', icon: <HomeIcon /> },
      { text: 'About', icon: <HomeIcon /> },
      { text: 'Contact', icon: <HomeIcon /> },
      { text: 'Home', icon: <HomeIcon /> },
      { text: 'About', icon: <HomeIcon /> },
      { text: 'Contact', icon: <HomeIcon /> },
    ]}
    {...args}
  />
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});

Default.args = {
  iconSize: '16px',
  fontSize: '18px',
  maxWidth: '500px',
  activeUnderline: true,
  activecolor: '#353840',
  iconColor: '#707a83',
  color: '#707a83',
  scrollThumbColor: '#707a83',
  item: [
    { text: 'Home', icon: <HomeIcon /> },
    { text: 'About', icon: <HomeIcon /> },
    { text: 'Contact', icon: <HomeIcon /> },
    { text: 'Home', icon: <HomeIcon /> },
    { text: 'About', icon: <HomeIcon /> },
    { text: 'Contact', icon: <HomeIcon /> },
    { text: 'Home', icon: <HomeIcon /> },
    { text: 'About', icon: <HomeIcon /> },
    { text: 'Contact', icon: <HomeIcon /> },
    { text: 'Home', icon: <HomeIcon /> },
    { text: 'About', icon: <HomeIcon /> },
    { text: 'Contact', icon: <HomeIcon /> },
  ],
};
