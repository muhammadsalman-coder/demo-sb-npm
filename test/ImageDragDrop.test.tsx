import React from 'react';
import { Default as SalmanImageDragDrop } from '../stories/SalmanImageDragDrop.stories';

import { render } from '@testing-library/react';

describe('SalmanImageDragDrop test', () => {
  it('renders', () => {
    const { baseElement } = render(<SalmanImageDragDrop />);
    expect(baseElement).toMatchSnapshot();
  });
});
