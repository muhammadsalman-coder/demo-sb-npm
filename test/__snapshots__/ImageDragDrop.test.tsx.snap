// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SalmanImageDragDrop test renders 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-glnqz7"
    >
      <input
        accept="image/png, image/jpg, image/jpeg"
        type="file"
      />
      <div
        class="MuiBox-root css-0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CameraAltIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="3.2"
          />
          <path
            d="M9 2 7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
          />
        </svg>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-yk42ob-MuiTypography-root"
        >
          Drag and Drop Image Here or
          <span
            class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
          >
            Click to Browse
          </span>
        </p>
        <ul>
          <li>
            <p>
              JPG, PNG
            </p>
          </li>
          <li>
            <p>
              Image will be cropped in 3:1 ratio
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;
